client/components/emojify/index.jsx

```diff
@@ -34,8 +38,14 @@ export default class Emojify extends PureComponent {
 	}
 
 	render() {
+		if ( 'string' === typeof this.props.children ) {
```

Comment:
See https://facebook.github.io/react/docs/jsx-in-depth.html#jsx-children. There are so many options out there what can be passed as a component's children property. You could possibly pass a component which produces string, but I guess it would be recognized as an object.